Merge https://github.com/k8snetworkplumbingwg/ovs-cni:main into main

SchSeba opened the pull request.

oshoval committed:
cluster: Allow kubevirtci tag overriding (#319)

* cluster: Allow kubevirtci tag overriding

It is helpful either manually or for example
when used from CNAO, to use same tag of the cluster that CNAO deployed.
Otherwise a kubevirtci tag mismatch can cause failures once running from CNAO.

Signed-off-by: Or Shoval <oshoval@redhat.com>

* dockerfile: Fix go downloading

Previous method doesnt support newer versions anymore
(or not at least atm).

Just use wget instead.

Signed-off-by: Or Shoval <oshoval@redhat.com>

---------

Signed-off-by: Or Shoval <oshoval@redhat.com>

oshoval committed:
Fix bump-kubevirtci.sh (#321)

Since line format was changed to allow overriding default value,
need to adapt the bump script.

Signed-off-by: Or Shoval <oshoval@redhat.com>

phoracek committed:
Bump Kubernetes dependencies to 1.30.3

Signed-off-by: Petr Horacek <phoracek@redhat.com>

phoracek committed:
Merge pull request #323 from phoracek/kube-1.30.3

Chore: Bump Kubernetes dependencies to 1.30.3

kim-tae-kyung committed:
Add support for userspace device drivers with HW offload mode (#322)

* Add support for userspace drivers with HW offload mode

Add support for SR-IOV VFs using both hardware offload (switchdev)
and userspace device driver such as vfio-pci.

Signed-off-by: Taekyung Kim <kim.tae.kyung@sk.com>

* Update MAC address if provided in HW offload mode

If the MAC address is provided from args, update the MAC address of the
VF to the provided MAC address via netlink.

Signed-off-by: Taekyung Kim <kim.tae.kyung@sk.com>

---------

Signed-off-by: Taekyung Kim <kim.tae.kyung@sk.com>

phoracek committed:
Have renovate update dependencies (#325)

Signed-off-by: Petr Horacek <phoracek@redhat.com>

redhat-renovate-bot committed:
chore(deps): update all dependencies on main (main) (#326)

* chore(deps): update all dependencies on main

Signed-off-by: redhat-renovate-bot <redhat-internal-renovate@redhat.com>

* chore(deps): Fix import paths after bump

Some of the imported modules has changed their paths since they were
introduced to the project.

This commit reflects their current path.

Without this change, go mod tidy fails.

Signed-off-by: Petr Horacek <phoracek@redhat.com>

* Fix ovsdb signatures after bump

OVSDB has changed some of its signatures. This commit reflects that
to allow compilation of ovsdb.go.

Signed-off-by: Petr Horacek <phoracek@redhat.com>

* Install Go detected from go.mod

Replace duplicated Go version requirement from the script and use the
one specified in the go.mod instead.

Without this change, linter fails on OOM due to incompatible version:
running lint of 1.22 codebase with 1.21 Go.

Signed-off-by: Petr Horacek <phoracek@redhat.com>

* Bump version of the go linter

Signed-off-by: Petr Horacek <phoracek@redhat.com>

* chore(deps): Fix cni testlib singnatures after bump

Some of the CNI testlib functions has changed their signatures. This
commits reflects that in our unit tests.

Signed-off-by: Petr Horacek <phoracek@redhat.com>

* chore(deps): Switch to ginkgo v2

The QE reporter now seems to rely on Ginkgo v2. Using Ginkgo v1 results
in a compilation failure.

This commit bumps the used Ginkgo version to v2.

Signed-off-by: Petr Horacek <phoracek@redhat.com>

* Fix linter deprecation warnings

With the recent bump, some of the importent functions became
deprecated.

This commit replaces deprecated functions with their replacements.

Signed-off-by: Petr Horacek <phoracek@redhat.com>

* Remove minimal Go requirement from SR-IOV

This explicit requirement for Go >=1.10 is ancient and not needed
anymore since all the compilation is happening with 1.22+ already.

This requirement also started preventing compilation, thus removing it.

Signed-off-by: Petr Horacek <phoracek@redhat.com>

* Explicitly bring container veth up

containernetwork/plugins changed the behavior of SetupVeth in 1.1,
to not set the container-side of the veth up.

This commit makes sure we set it up after the veth is created.

Signed-off-by: Petr Horacek <phoracek@redhat.com>

---------

Signed-off-by: redhat-renovate-bot <redhat-internal-renovate@redhat.com>
Signed-off-by: Petr Horacek <phoracek@redhat.com>
Co-authored-by: Petr Horacek <phoracek@redhat.com>

SchSeba committed:
Allow to configure golang flags (#329)

Signed-off-by: Sebastian Sch <sebassch@gmail.com>

zeeke commented:
/lgtm
/approve

zeeke commented:
I believe we need merge automation for this:
- https://github.com/openshift/release/pull/56506

tabuhariri committed:
pull e2e ovs-cni fix: install latest ginkgo (#334)

getting this error in the ci job when trying to pull e2e ovs-cni:
go: 'go install' requires a version when current directory is not in a
module

Signed-off-by: Tarek Abu-Hariri <tabuhariri@nvidia.com>

phoracek committed:
Bump go to 1.22.9 (#332)

Signed-off-by: Petr Horacek <phoracek@redhat.com>

tabuhariri committed:
Add TCP connection support for OVS (#330)

- Extend OVS connection options beyond Unix sockets
- Implement TCP endpoint configuration for marker binary
- Use -ovs-socket flag with format "tcp:<IP>:<PORT>"

Example usage: -ovs-socket tcp:127.0.0.1:6640

**What this PR does / why we need it**:

Add Support for ovs tcp connection

**Special notes for your reviewer**:

**Release note**:

```release-note
This feature allows for remote OVS connections, enhancing flexibility in network configurations.
User will be allowed to use tcp connection to OVS by specifying OVS
socket endpoint by using -ovs-socket flag with tcp endpoint "tcp:<IP>:<PORT>".
```

Signed-off-by: Tarek Abu-Hariri <tabuhariri@nvidia.com>

ykulazhenkov committed:
Fix: multiarch build by using crosscompilation (#336)

Use quay.io/centos/centos:stream9 with the –platform=$BUILDPLATFORM flag,
in this case the image is pulled for the builder host's current architecture.
The BUILDOS and BUILDARCH args are used to download the right go binary for
the build platform.
Cross-compilation occurs in the builder image using TARGETOS and TARGETARCH build arguments to determine the target OS/arch.
These args are set automatically by the multiarch-build process (with docker buildx).
The final container image (registry.access.redhat.com/ubi9/ubi-minimal) is pulled for the correct target architecture, such as amd64 or arm64.

This update fixes support for multiarch builds and speeds up image building, as cross-compilation is faster than compiling on a non-native platform.

Signed-off-by: Yury Kulazhenkov <ykulazhenkov@nvidia.com>

ashokpariya0 committed:
Add support for s390x architecture in GitHub Actions workflows. (#337)

Updated GitHub Actions workflows to include linux/s390x in BUILD_PLATFORMS.
Modified hack/install-go.sh to dynamically select the appropriate Go tarball
based on the system's architecture(amd64, arm64, or others), allowing better
support for multi-architecture environments.

Signed-off-by: Ashok Pariya <ashok.pariya@ibm.com>

jiayoukun committed:
Add the Pod's UID to the external_ids field of OVSDB (#311)

**What this PR does / why we need it**:

By adding the UID of a Pod to the external_ids field of the
corresponding veth interface in the OVSDB, it becomes convenient for the
Pod object to query the OVSDB using the UID to find the vethName of the
Pod on the host machine, and then deliver OVS flow tables accordingly.

Currently, the host machine's veth interface in the OVSDB only stores
the Netns field, which is useful, but for Pod objects, their netns is
not publicly accessible. This change addresses this limitation by
allowing Pod objects to query the OVSDB directly using their UID.

**Special notes for your reviewer**:

**Release note**:

```release-note
Added the Pod UID to the external_ids field of the corresponding veth
interface in the OVSDB.
This change allows querying the vethName of a Pod on the host machine
using its UID, improving the ability to deliver OVS flow tables for Pod
objects.
```

Signed-off-by: jiayoukun <824807548@qq.com>
Co-authored-by: liangkr <liangkr@yusur.tech>

SchSeba committed:
Merge https://github.com/k8snetworkplumbingwg/ovs-cni:main into main

zeeke commented:
/lgtm
